.github/workflows/_e2e.yaml

```diff
@@ -103,6 +108,17 @@ jobs:
           if [[ "${{ env.E2E_PARAM_TARGET }}" == "multizone" ]]; then
             export KUMA_DEFAULT_RETRIES=60
           fi
+          
+          function on_exit()
+          {
+            docker logout docker.io
+          }
+          K3D_REGISTRY_FILE=$(tools/ci/prepare-pull-credentials.sh)
+          if [[ "$K3D_REGISTRY_FILE" != "" ]]; then
+            export K3D_REGISTRY_FILE
+            trap "on_exit" EXIT
+          fi
```

Comment:
What I mean is that instead of calling this here explicitly it could just be part of the Makefile. With targets just like `kind/load/images` and `k3d/load/images`. For .e.g: `k3d/setup-docker-credentials` and `kind/setup-docker-credentials` it's closer to what's done everywhere